hack/ebs-scale-test/helpers/scale-test/snapshot-volume-scale-test/snapshot-volume-scale.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# We expect this helper script is sourced from hack/ebs-scale-test
+path_to_snapshot_test_dir="${BASE_DIR}/helpers/scale-test/snapshot-volume-scale-test"
+
+export SNAPSHOTS_PER_VOLUME=${SNAPSHOTS_PER_VOLUME:=1}
+
+snapshot_scale_test() {
+  manifest_path="$path_to_snapshot_test_dir/snapshot-volume-scale.yaml"
+  export_manifest_path="$EXPORT_DIR/snapshot-manifest.yaml"
+
+  echo "Applying $manifest_path. Exported to $export_manifest_path"
+  gomplate -f "$manifest_path" -o "$export_manifest_path"
+  kubectl apply -f "$export_manifest_path"
+
+  trap 'echo "Test interrupted! Deleting test resources to prevent leak"; cleanup_snapshot_resources' EXIT
+
+  echo "Waiting for all PVCs to be bound"
+  wait_for_pvcs_bound
+
+  echo "Creating $REPLICAS volume snapshots ($SNAPSHOTS_PER_VOLUME per volume)"
+  create_snapshots
+
+  echo "Waiting for snapshots to be ready"
+  wait_for_snapshots_ready
+
+  echo "Restoring volumes from snapshots"
+  restore_volumes_from_snapshots
+
+  echo "Waiting for restored PVCs to be bound"
+  wait_for_restored_pvcs_bound
+
+  echo "Deleting snapshots"
+  delete_snapshots
+
+  echo "Waiting for VolumeSnapshotContents to be deleted"
+  wait_for_volume_snapshot_contents_deleted
+
+  echo "Deleting PVCs"
+  delete_pvcs
+
+  echo "Waiting for PVs to be deleted"
+  wait_for_pvs_deleted
+
+  echo "Cleaning up snapshot resources"
+  cleanup_snapshot_resources
+
+  trap - EXIT
```

Comment:
What's the point of this?

Reply:
This clears the previously set trap:
```
trap 'echo "Test interrupted! Deleting test resources to prevent leak"; cleanup_snapshot_resources' EXIT
```

It's a common pattern in cleanup scripts to ensure resources are properly cleaned up even if the script fails, the trap is removed once everything has completed successfully.